src/1-Profile/Edit-Profile.tsx

```diff
@@ -30,66 +31,63 @@ const EditProfile = ({navigation}:StackNavigationProps):JSX.Element => {
     const RequestAccountHeader = {
       headers: {
         "jwt": jwt, 
-        "userID": userID,
       }
     }
-    // store user profile data as an indexable object
-    const userProfileValues: Record<string, string> = {};
-    for (const [key, value] of Object.entries(userProfile)) {
-      userProfileValues[key] = value;
-    }
 
-    // assign user profile data as default values for fields
-    const inputFieldJSON: Record<string, string> = {};
-    EDIT_PROFILE_FIELDS.forEach((field) => {
-      if (userProfileValues[field.field]) inputFieldJSON[field.field] = userProfileValues[field.field];
-    })
+    const createFormValues = ():Record<string, string> => {
+      const formValues: Record<string, string> = {};
+      EDIT_PROFILE_FIELDS.forEach((field) => {
+        //@ts-ignore
+        formValues[field.field] = userProfile[field.field];
+      });
+      return formValues;
+    }
 
     const {
       control,
       handleSubmit,
       formState: { errors },
       clearErrors,
     } = useForm({
-      defaultValues: inputFieldJSON
+      defaultValues: createFormValues()
     });
 
-    const onEditFields = (fieldData:any):void => {
+    const onEditFields = (formValues:Record<string, string>):void => {
 
       // filter out fields that didn't change
-      const editedFields: Record<string, string> = {};
-      for (const [key, value] of Object.entries(fieldData)) {
-        if (value !== userProfileValues[key]) editedFields[key] = value as unknown as string;
+      const editedFields = {} as ProfileEditRequestBody;
+      for (const [key, value] of Object.entries(formValues)) {
+        //@ts-ignore
+        if (value !== userProfile[key]) editedFields[key] = value;
       }
-
       // send data to server
       axios.patch(`${DOMAIN}/api/user/` + userProfile.userID, editedFields, RequestAccountHeader,
         ).then(response => {
             console.log("Profile edit success.");
 
-            // IMPORTANT: The following assumes that this route will return the user's userRole, userRoleList, and walkLevel by default, and other fields that changed in addition.
+            // IMPORTANT: The following assumes that this route will return basic userProfile params (the user's userRole, userRoleList, and walkLevel by default) and other fields that changed in addition.
             // IF THIS CHANGES, EDIT PROFILE WILL BREAK!
             if (Object.values(response.data).length > 3) {
+
```

Comment:
I didn't catch the response only returns a partial profile.  I'm going to correct the server to return the full profile just like signup  POST response currently does.  I think leaving the filter won't harm future changes.

- This length greater than three seems unreliable in the future.
- Can you explain why it's so fragile if the response changes?